source/Calamari.Shared/Integration/Packages/GenericPackageExtractorFactory.cs

```diff
@@ -13,14 +14,11 @@ public GenericPackageExtractor createStandardGenericPackageExtractor()
             return new GenericPackageExtractor();
         }
 
-        public GenericPackageExtractor createJavaGenericPackageExtractor(ICalamariFileSystem fileSystem)
```

Comment:
Rather than taking dependency on calamari filesytem, building commands below etc, we can simply pass down a class that proxies all the calls down into the Java Tool